diff --git a/internal/compile_libs.sh b/internal/compile_libs.sh
index 38c6655..49a6a7e 100755
--- a/internal/compile_libs.sh
+++ b/internal/compile_libs.sh
@@ -27,7 +27,7 @@ if [ -n "${CPPFLAGS}" ]; then
   echo "CPPFLAGS=${CPPFLAGS}"
 fi
 
-g++ $CFLAGS $CPPFLAGS $CXXFLAGS -shared -fPIC \
+g++ $CFLAGS $CPPFLAGS $CXXFLAGS -dynamiclib -fPIC \
   cldutil.cc cldutil_shared.cc compact_lang_det.cc compact_lang_det_hint_code.cc \
   compact_lang_det_impl.cc  debug.cc fixunicodevalue.cc \
   generated_entities.cc  generated_language.cc generated_ulscript.cc  \
@@ -37,17 +37,5 @@ g++ $CFLAGS $CPPFLAGS $CXXFLAGS -shared -fPIC \
   cld_generated_cjk_delta_bi_4.cc generated_distinct_bi_0.cc  \
   cld2_generated_quadchrome_2.cc cld2_generated_deltaoctachrome.cc \
   cld2_generated_distinctoctachrome.cc  cld_generated_score_quad_octa_2.cc  \
-  -o libcld2.so $LDFLAGS -Wl,-soname=libcld2.so
-
-g++ $CFLAGS $CPPFLAGS $CXXFLAGS -shared -fPIC \
-  cldutil.cc cldutil_shared.cc compact_lang_det.cc compact_lang_det_hint_code.cc \
-  compact_lang_det_impl.cc  debug.cc fixunicodevalue.cc \
-  generated_entities.cc  generated_language.cc generated_ulscript.cc  \
-  getonescriptspan.cc lang_script.cc offsetmap.cc  scoreonescriptspan.cc \
-  tote.cc utf8statetable.cc  \
-  cld_generated_cjk_uni_prop_80.cc cld2_generated_cjk_compatible.cc  \
-  cld_generated_cjk_delta_bi_32.cc generated_distinct_bi_0.cc  \
-  cld2_generated_quad0122.cc cld2_generated_deltaocta0122.cc \
-  cld2_generated_distinctocta0122.cc  cld_generated_score_quad_octa_0122.cc  \
-  -o libcld2_full.so $LDFLAGS -Wl,-soname=libcld2_full.so
+  -o libcld2.dylib $LDFLAGS -Wl,-install_name,libcld2.dylib
